samples/directx/d3d10_opencl.cpp

```diff
@@ -0,0 +1,397 @@
+
+                    D3D10_MAPPED_TEXTURE2D mappedTex;
+                    r = m_pSurface->Map(subResource, D3D10_MAP_WRITE_DISCARD, 0, &mappedTex);
+                    if (FAILED(r))
+                    {
+                        return r;
+                    }
+
+                    cv::Mat m(m_height, m_width, CV_8UC4, mappedTex.pData, (int)mappedTex.RowPitch);
+
+                    if (!m_disableProcessing)
+                    {
+                        // blur D3D10 surface with OpenCV on CPU
+                        cv::blur(m, m, cv::Size(15, 15), cv::Point(-7, -7));
+                    }
+
+                    m_pSurface->Unmap(subResource);
+
+                    break;
+                }
+
+                case 2:
+                {
+                    // process video frame on GPU
+                    cv::UMat u;
+
+                    cv::directx::convertFromD3D10Texture2D(pSurface, u);
+
+                    if (!m_disableProcessing)
+                    {
+                        // blur D3D9 surface with OpenCV on GPU with OpenCL
+                        cv::blur(u, u, cv::Size(15, 15), cv::Point(-7, -7));
+                    }
+
+                    cv::directx::convertToD3D10Texture2D(u, pSurface);
+
+                    break;
+                }
+
+            } // switch
+
+            print_info(pSurface, m_mode, getFps(), m_oclDevName);
+
+            // traditional DX render pipeline:
+            //   BitBlt surface to backBuffer and flip backBuffer to frontBuffer
+            m_pD3D10Dev->CopyResource(m_pBackBuffer, pSurface);
+
+            // present the back buffer contents to the display
+            // switch the back buffer and the front buffer
+            r = m_pD3D10SwapChain->Present(0, 0);
+            if (FAILED(r))
+            {
+                return -1;
+            }
+        }
+
+        catch (cv::Exception& e)
+        {
+            std::cerr << "Exception: " << e.what() << std::endl;
+            return 10;
+        }
+
+        return 0;
+    }
+
+    int cleanup(void)
+    {
+        SAFE_RELEASE(m_pSurface);
+        SAFE_RELEASE(m_pBackBuffer);
+        SAFE_RELEASE(m_pD3D10SwapChain);
+        SAFE_RELEASE(m_pRenderTarget);
+        SAFE_RELEASE(m_pD3D10Dev);
+        return 0;
+    }
+
+private:
+    bool                    m_shutdown;
+    int                     m_mode;
+    cv::String              m_modeStr[3];
+    int                     m_disableProcessing;
+    ID3D10Device*           m_pD3D10Dev;
+    IDXGISwapChain*         m_pD3D10SwapChain;
+    ID3D10Texture2D*        m_pBackBuffer;
+    ID3D10Texture2D*        m_pSurface;
+    ID3D10RenderTargetView* m_pRenderTarget;
+    cv::VideoCapture        m_cap;
+    cv::Mat                 m_frame_bgr;
+    cv::Mat                 m_frame_rgba;
+    cv::ocl::Context        m_oclCtx;
+    cv::String              m_oclPlatformName;
+    cv::String              m_oclDevName;
+};
+
+
+using namespace cv;
```

Comment:
May be place this line in the beginning of file (or move to some sample header)?